### PR TITLE
refactor: `{count, exists, forall}` to be effect polymorphic

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to #3798

### CODE REVIEW

File: main/src/library/String.flix [hunk 1]
```diff
@@ -328,7 +328,7 @@ namespace String {
     /// Returns `None` if no character in `s` satisfies `f`.
     ///
     @Time(time(f) * length(s)) @Space(space(f))
-    pub def findIndexOfLeft(f: Char -> Bool, s: String): Option[Int32] =
+    pub def findIndexOfLeft(f: Char -> Bool & ef, s: String): Option[Int32] & ef =
```
Comment (ninaandrup) [inline main/src/library/String.flix:331]: `String.forall` cannot be effect polymorphic, if `findIndexOfLeft` is not effect polymorphic too (or cast). Should `findIndexOfLeft` be effect polymorphic?

Comment (magnus-madsen) [reply main/src/library/String.flix:331]: Yes, lets make find effect polymorphic too.

**Comment by magnus-madsen** [comment]
@ninaandrup I think one of Jakob's PRs caused a conflict. Sorry about that.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**